src/fatx_fat.c

```diff
@@ -20,12 +20,118 @@
 #include <stdbool.h>
 #include "fatx_internal.h"
 
+#define MAX_WRITE_SIZE 1024 * 1024 * 8 // 8mb
+
 static bool fatx_cluster_valid(struct fatx_fs *fs, size_t cluster)
 {
-    return (cluster >= FATX_FAT_RESERVED_ENTRIES_COUNT) &&
+    return (cluster >= 0) &&
```

Comment:
This might be a controversial change which is which I am leaving a comment about it. Without this, `fatx_write_fat(...)` will fail in `fatx_init_root(...)` when trying to write 0xFFF8 to the root directory cluster id (0).